backend/main.py

```diff
@@ -0,0 +1,174 @@
+import sqlite3, uuid, json, os
+from bottle import Bottle, run, request, response, HTTPResponse
+
+
+from base64 import (
+    b64encode,
+    b64decode,
+)
+
+from Crypto.Hash import SHA256
+from Crypto.Signature import pkcs1_15
+from Crypto.PublicKey import RSA
+
+app = Bottle()
+
+# DB with SQLite
+con = sqlite3.connect("users.db")
+cur = con.cursor()
+
+def init():
+
+    def createDB():
+        cur.execute("""
+        CREATE TABLE IF NOT EXISTS users (
+        id VARCHAR(36),
+        username VARCHAR(30),
+        location VARCHAR(30),
+        PRIMARY KEY (id));""")
+        con.commit()
+        cur.execute("""
+        CREATE TABLE IF NOT EXISTS friendReq (
+        friendA VARCHAR(36),
+        friendB VARCHAR(36),
+        accept BIT DEFAULT 0);""")
+        con.commit()
+
+    createDB()
+
+
+
+# route for debugging
+# get information about all users
+@app.route('/api/call/getallusers')
+@app.route('/api/call/getAllUsers')
+def api_getAllUsers():
+
+    def getAllUsers():
+        sql_query = """SELECT * FROM users"""
+        cur.execute(sql_query)
+        content=cur.fetchall()
+        erg = {}
+        for e in content:
+            erg[e[0]] = {"username":e[1],"location":e[-1]}
+        return json.dumps(erg)
+
+    response.headers['Content-Type'] = 'application/json'
+    return getAllUsers()
+
+
+# route for user registration
+# without public key - tbd
+@app.route('/api/call/createuser')
+@app.route('/api/call/createUser')
```

Comment:
Use POST

Reply:
how does the client get the uuid from the server?